lib/paperclip/io_adapters/http_url_proxy_adapter.rb

```diff
@@ -9,7 +9,13 @@ def self.register
     REGEXP = /\Ahttps?:\/\//
 
     def initialize(target, options = {})
-      super(URI(URI.escape(target)), options)
+      already_escaped = CGI.unescape(target) != target
+      escaped_target = if already_escaped
+                         target
+                       else
+                         URI.escape(CGI.unescape(target))
```

Comment:
`CGI.escape` or `URI.encode_www_form_component` escapes full URI, even protocol what makes URI understand it as a file path instead of a URL doe to `URI.regexp`.